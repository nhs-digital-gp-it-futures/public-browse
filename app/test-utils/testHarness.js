import express from 'express';
import nunjucks from 'nunjucks';
import { App } from '../../app';

export const testHarness = () => {
  const app = new App().createApp();
  const router = express.Router();

  return {
    createComponentDummyApp(template, context) {
      const dummyRouter = router.get('/', (req, res) => {
        res.render(template, context);
      });
      app.use(dummyRouter);
      return app;
    },
    createTemplateDummyApp(template, context) {
      const dummyRouter = router.get('/', (req, res) => {
        const viewToTest = nunjucks.renderString(template, context);
        res.send(viewToTest);
      });
      app.use(dummyRouter);
      return app;
    },
  };
};
