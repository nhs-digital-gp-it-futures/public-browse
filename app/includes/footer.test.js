import { componentTester, snapshotTest } from '../test-utils/componentTester';

const setup = {
  template: {
    path: 'includes/footer.njk',
  },
};

const footerLinks = [
  {
    label: 'Buyer\'s Guide',
    URL: '/guide',
  },
  {
    label: 'NHS Digital Helpdesk',
    URL: '/guide#contact-us',
  },
  {
    label: 'NHS Digital',
    URL: 'https://digital.nhs.uk/',
  },
  {
    label: 'About GP IT Futures',
    URL: 'https://digital.nhs.uk/services/future-gp-it-systems-and-services',
  },
  {
    label: 'Capabilities & Standards Model',
    URL: 'https://gpitbjss.atlassian.net/wiki/spaces/GPITF/overview',
  },
];

describe('footer', () => {
  it('should render the footer panel', componentTester(setup, (harness) => {
    const context = { footerLinks };

    harness.request(context, ($) => {
      const snapshot = snapshotTest($, '[data-test-id="footer-component"]');
      expect(snapshot).toMatchSnapshot();
    });
  }));

  it('should render the footer legal panel', componentTester(setup, (harness) => {
    const context = { showLegalPane: false };

    harness.request(context, ($) => {
      const snapshot = snapshotTest($, '[data-test-id="legal-panel"]');
      expect(snapshot).toMatchSnapshot();
    });
  }));
});
