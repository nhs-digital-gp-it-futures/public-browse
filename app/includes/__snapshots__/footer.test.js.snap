// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`footer should render the footer legal panel 1`] = `
"<div data-test-id=\\"legal-panel\\" class=\\"nhsuk-u-padding-bottom-3 nhsuk-width-container\\">
        <span class=\\"nhsuk-u-margin-right-2\\">Legal</span>
        <span class=\\"nhsuk-u-margin-right-2\\"><a href=\\"https://digital.nhs.uk/about-nhs-digital/privacy-and-cookies\\">Privacy and Cookies</a></span>
        <span class=\\"nhsuk-u-margin-right-2\\"><a href=\\"/document/terms-of-use.pdf\\">Terms and Conditions</a></span>
      </div>"
`;

exports[`footer should render the footer panel 1`] = `
"<div data-test-id=\\"footer-component\\">
      <footer role=\\"contentinfo\\">
  <div class=\\"nhsuk-footer\\" id=\\"nhsuk-footer\\">
    <div class=\\"nhsuk-width-container\\">
      <h2 class=\\"nhsuk-u-visually-hidden\\">Support links</h2>
      <ul class=\\"nhsuk-footer__list nhsuk-footer__list--three-columns\\">
        <li class=\\"nhsuk-footer__list-item\\"><a class=\\"nhsuk-footer__list-item-link\\" href=\\"/guide\\">Buyer's Guide</a></li>
        <li class=\\"nhsuk-footer__list-item\\"><a class=\\"nhsuk-footer__list-item-link\\" href=\\"/guide#contact-us\\">NHS Digital Helpdesk</a></li>
        <li class=\\"nhsuk-footer__list-item\\"><a class=\\"nhsuk-footer__list-item-link\\" href=\\"https://digital.nhs.uk/\\">NHS Digital</a></li>
        <li class=\\"nhsuk-footer__list-item\\"><a class=\\"nhsuk-footer__list-item-link\\" href=\\"https://digital.nhs.uk/services/future-gp-it-systems-and-services\\">About GP IT Futures</a></li>
        <li class=\\"nhsuk-footer__list-item\\"><a class=\\"nhsuk-footer__list-item-link\\" href=\\"https://gpitbjss.atlassian.net/wiki/spaces/GPITF/overview\\">Capabilities &amp; Standards Model</a></li>
      </ul>
      
      <p class=\\"nhsuk-footer__copyright\\">Â© Crown copyright</p>
    </div>
  </div>
</footer>

    </div>"
`;
